src/hx/libs/regexp/RegExp.cpp

```diff
@@ -165,11 +190,12 @@ Dynamic _hx_regexp_new_options(String s, String opt)
    #ifdef HX_SMART_STRINGS
    if (s.isUTF16Encoded())
    {
-      const char *error = 0;
-      int err_offset = 0;
-      pcre16 *p =  pcre16_compile((PCRE_SPTR16)s.raw_wptr(),options|PCRE_UTF16,&error,&err_offset,NULL);
-      if( !p )
-         hx::Throw( HX_CSTRING("Regexp compilation error : ")+String(error)+HX_CSTRING(" in ")+s);
+      int error_code;
+      size_t error_offset;
+      pcre2_code_16 *p = pcre2_compile_16((PCRE2_SPTR16)s.raw_wptr(),s.length,options,&error_code,&error_offset,NULL);

```

Comment:
Same thing here: is `s.length` the byte length or the code unit length?

Reply:
Here it's all good, since we aren't doing any conversions:
```haxe
// with hxcpp_smart_strings (UTF-16)
trace("🙂".length); // 2
// without hxcpp_smart_strings (UTF-8)
trace("🙂".length); // 4
```